### PR TITLE
Stop repairing bad relevances in the kernel

### PR DESCRIPTION
**Pull request by SkySkimmer**
It has been default-error for a while, nobody should be relying on it.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
full CI https://gitlab.inria.fr/coq/coq/-/pipelines/987636 was green before I added the changelog

File: pretyping/typing.ml [hunk 1]
```diff
@@ -403,13 +403,20 @@ let judge_of_array env sigma u tj defj tyj =
...
 (* no need for default printer, pretyping is always linked with himsg in practice *)
 
 let warn_bad_relevance_binder ?loc env sigma rlv bnd =
-  match CWarnings.warning_status Typeops.bad_relevance_warning with
+  match CWarnings.warning_status bad_relevance_warning with
 | CWarnings.Disabled | CWarnings.Enabled ->
-  CWarnings.warn bad_relevance_msg ?loc (env,sigma,Typeops.BadRelevanceBinder (rlv, bnd))
+  CWarnings.warn bad_relevance_msg ?loc (env,sigma,BadRelevanceBinder (rlv, bnd))
 | CWarnings.AsError ->
   CErrors.anomaly (CErrors.print (PretypeError (env, sigma, TypingError (Type_errors.BadBinderRelevance (rlv, bnd)))))
```
Comment (silene) [inline pretyping/typing.ml:421]: I am really confused by this function. Why is it an anomaly and not just an error? Does that mean that the user cannot trigger it (but a plugin could)? Also, what is the rationale for allowing the user to disable an anomaly?

Comment (ppedrot) [reply pretyping/typing.ml:421]: I agree this should be an error. The convention in the kernel is not consistent, sometimes we throw an error at ill-formed terms, sometimes we fail. I think that given that this code can be called by the checker, an anomaly should really happen only when there is a programming error, not when the kernel gets some ill-formed input.

**Comment by ppedrot** [comment]
I think we should postpone this to 8.21, the branch is near and if anything this will make working around SProp bugs harder.

File: pretyping/typing.ml (hunk 1 above)
Comment (SkySkimmer) [reply pretyping/typing.ml:421]: I have no memory of making this an anomaly but blame says it was me.

@ppedrot note that this is pretyping/typing not the kernel.

Comment (ppedrot) [reply pretyping/typing.ml:421]: (My remark also goes for the upper layers, especially since we have Ltac2 to poke around terms and break invariants.)

Comment (SkySkimmer) [reply pretyping/typing.ml:421]: https://github.com/coq/coq/pull/19195

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**